shared/util.go

```diff
@@ -819,6 +819,16 @@ func StringMapHasStringKey(m map[string]string, keys ...string) bool {
 	return false
 }
 
+// BlockFsUUID returns the UUID of a filesystem from the block device.
+func BlockFsUUID(pathName string) (string, error) {
```

Comment:
We should really only be moving things *out* of the bloated catch-call `shared` package and not into it.

As `lxd/resources` is already AGPL 3.0, there should be no problem with it consuming other packages that are outside of `shared`.

So I suggest moving this function into `lxd/storage/filesystem` as that seems more appropriate.

Reply:
It could also be in `lxd/storage/block` potentially.

And the "Fs" should be "FS" I think as its an acronym.